swarm/CHANGELOG.md

```diff
@@ -1,3 +1,8 @@
+## 0.44.2 - unreleased
+
+- Add `FromSwarm::NewExternalAddrOfPeer` event.
+  See [PR 4371](https://github.com/libp2p/rust-libp2p/pull/4371)
```

Comment:
Let's also document the new `Swarm` API please. That is what users will interact with most.

Reply:
Added more info, but I'm not sure if it's sufficient for someone using the library :thinking: